### Problem Statement 1:
In each of the following situations, state whether it is a correctly stated hypothesis
testing problem and why?
1. 𝐻0: 𝜇 = 25, 𝐻1: 𝜇 ≠ 25
2. 𝐻0: 𝜎 > 10, 𝐻1: 𝜎 = 10
3. 𝐻0: 𝑥 = 50, 𝐻1: 𝑥 ≠ 50
4. 𝐻0: 𝑝 = 0.1, 𝐻1: 𝑝 = 0.5
5. 𝐻0: 𝑠 = 30, 𝐻1: 𝑠 > 30

1. **yes**, since H0 is Hypothesis which is assumed initially to be true. On the other hand the alternative is statment that contradicts the Null Hypothesis

    𝐻1: 𝜇 ≠ 25, says that value of mean can be greater or less than 25, it is also called a **two-sided alternative hypothesis**

    In simple words we can say that we assume that our H0 is true while H1 is assuming that the value may be geater or less than the mean value


2. **No**, We always assume Null Hypothesis as equality claim. and Alternate Hypothesisas inquality claim, the above statement is simply inverse.


3. **No**, hypothesis are always statemnt about the population or distibution under study(i.e statistic) not on single data value or a prameter


4. **No**, values in both hypothesis are different and both hypothesis have equality


5. **No**, here it is making statemnt on sample s, but Hypothesis are always statement about the population or distribution under study, not statemnts about the sample

### Problem Statement 2:
The college bookstore tells prospective students that the average cost of its textbooks is Rs. 52 with a standard deviation of Rs. 4.50. 

A group of smart statistics students thinks that the average cost is higher. To test the bookstore’s claim against their  lternative, the students will select a random sample of size 100. Assume that the mean from their random sample is Rs. 52.80. Perform a hypothesis test at the 5% level of significance and state your decision.

<img src="img\p2.png">

In [1]:
import numpy as np
from scipy import stats
import seaborn as sns
import matplotlib.pyplot as plt
from scipy.stats import binom
import scipy.stats as st

In [10]:
# lets perform a Z-test
# 𝐻0: 𝜇 = 52, 𝐻1: 𝜇  ≠   52
actual_mean = 52
assumed_mean = 52.80
n = 100
std_dev = 4.50
alpha = 0.05
std_error = std_dev/np.sqrt(n)
z_score = (assumed_mean-actual_mean)/ std_error
print("Z_score : ",z_score)
area_above = 1-st.norm.cdf(z_score)
print("%area above {} is {}".format(z_score,area_above))
z_score1 = st.norm.ppf(area_above)
area_below = st.norm.cdf(z_score1)
print("%area below {} is {}".format(z_score1,area_below))
print("critical value(Z*) is {}, {} ".format(z_score1,z_score))

Z_score :  1.7777777777777715
%area above 1.7777777777777715 is 0.0377201798134007
%area below -1.777777777777772 is 0.03772017981340071
critical value(Z*) is -1.777777777777772, 1.7777777777777715 


In [11]:
total_prob = area_above+area_below
print("P(z ≠ 1.77) is ", total_prob)
if total_prob < alpha:
    print("Reject H0")
else:
    print("Fail to reject H0")

P(z ≠ 1.77) is  0.0754403596268014
Fail to reject H0


In [20]:
 st.norm.ppf(1-0.975)

-1.959963984540054

In [23]:
# Alternatively
#divide alpha as 2 tailed test
two_tail_alpha = alpha/2

# compute critcal value
# % area below .025 is 0.975
area = 1-two_tail_alpha
significance_positive_zscore = st.norm.ppf(area)
print("Z_value of {} is : {}".format(area,significance_positive_zscore))
significance_negative_zscore = st.norm.ppf(two_tail_alpha)
print("Z_value of {} is : {}".format(two_tail_alpha,significance_negative_zscore))
print("critical value of significance at 5% is ±{}".format(significance_positive_zscore))

# comparte critical score for accepting and rejecting null hypothesis
if significance_positive_zscore < z_score and significance_negative_zscore < z_score1 :
    print("Reject H0")
else:
    print("Fail to reject H0")

Z_value of 0.975 is : 1.959963984540054
Z_value of 0.025 is : -1.9599639845400545
critical value of significance at 5% is ±1.959963984540054
Fail to reject H0


### Problem Statement 3:
A certain chemical pollutant in the Genesee River has been constant for several years with mean μ = 34 ppm (parts per million) and standard deviation σ = 8 ppm. A group of factory representatives whose companies discharge liquids into the river is now claiming that they have lowered the average with improved filtration devices. A group of environmentalists will test to see if this is true at the 1% level of significance. Assume \ that their sample of size 50 gives a mean of 32.5 ppm.
Perform a hypothesis test at the 1% level of significance and state your decision.

<img src="img\p3.png">

In [29]:
# lets perform a Z-test
# 𝐻0: 𝜇 = 52, 𝐻1: 𝜇  ≠   52
actual_mean = 34
assumed_mean = 32.5
n = 50
std_dev = 8
alpha = 0.01
std_error = std_dev/np.sqrt(n)
z_score = (assumed_mean-actual_mean)/ std_error
print("Z_score : ",z_score)
area_above = 1-st.norm.cdf(z_score)
print("%area above {} is {}".format(z_score,area_above))
z_score1 = st.norm.ppf(area_above)
area_below = st.norm.cdf(z_score1)
print("%area below {} is {}".format(z_score1,area_below))
print("critical value(Z*) is {}, {} ".format(z_score1,z_score))

Z_score :  -1.3258252147247767
%area above -1.3258252147247767 is 0.9075512005172
%area below 1.325825214724777 is 0.9075512005172
critical value(Z*) is 1.325825214724777, -1.3258252147247767 


In [30]:
total_prob = area_above+area_below
print("P(z ≠ 1.325) is ", total_prob)
if total_prob < alpha:
    print("Reject H0")
else:
    print("Fail to reject H0")

P(z ≠ 1.325) is  1.8151024010344
Fail to reject H0


In [31]:
# Alternatively
#divide alpha as 2 tailed test
two_tail_alpha = alpha/2

# compute critcal value
# % area below .025 is 0.975
area = 1-two_tail_alpha
significance_positive_zscore = st.norm.ppf(area)
print("Z_value of {} is : {}".format(area,significance_positive_zscore))
significance_negative_zscore = st.norm.ppf(two_tail_alpha)
print("Z_value of {} is : {}".format(two_tail_alpha,significance_negative_zscore))
print("critical value of significance at 5% is ±{}".format(significance_positive_zscore))

# comparte critical score for accepting and rejecting null hypothesis
if significance_positive_zscore < z_score and significance_negative_zscore < z_score1 :
    print("Reject H0")
else:
    print("Fail to reject H0")

Z_value of 0.995 is : 2.5758293035489004
Z_value of 0.005 is : -2.575829303548901
critical value of significance at 5% is ±2.5758293035489004
Fail to reject H0


### Problem Statement 4:
Based on population figures and other general information on the U.S. population, suppose it has been estimated that, on average, a family of four in the U.S. spends about $1135 annually on dental expenditures. Suppose further that a regional dental association wants to test to determine if this figure is accurate for their area of country. To test this, 22 families of 4 are randomly selected from the population in that area of the country and a log is kept of the family’s dental expenditure for one year. The resulting data are given below. Assuming, that dental expenditure is normally distributed in the population, use the data and an alpha of 0.5 to test the dental association’s hypothesis.

1008, 812, 1117, 1323, 1308, 1415, 831, 1021, 1287, 851, 930, 730, 699, 872, 913, 944, 954, 987, 1695, 995, 1003, 994

<img src="img\p4.png">

In [33]:
sample = [1008, 812, 1117, 1323, 1308, 1415, 831, 1021, 1287, 851, 930, 730, 699,
872, 913, 944, 954, 987, 1695, 995, 1003, 994]
sample_mean = np.sum(sample)/len(sample)

In [34]:
std_dev = np.std(sample)

In [42]:
# lets perform a Z-test
# 𝐻0: 𝜇 = 1135, 𝐻1: 𝜇  ≠   1135
actual_mean = 1135
assumed_mean = sample_mean
n = len(sample)
alpha = 0.05
std_error = std_dev/np.sqrt(n)
z_score = (assumed_mean-actual_mean)/ std_error
print("Z_score : ",z_score)
area_above = st.norm.cdf(z_score)
print("%area above {} is {}".format(z_score,area_above))
z_score1 = st.norm.ppf(1-area_above)
area_below = 1-st.norm.cdf(z_score1)
print("%area below {} is {}".format(z_score1,area_below))
print("critical value(Z*) is {}, {} ".format(z_score1,z_score))

Z_score :  -2.070747228595759
%area above -2.070747228595759 is 0.019191212063281355
%area below 2.070747228595759 is 0.019191212063281338
critical value(Z*) is 2.070747228595759, -2.070747228595759 


In [43]:
total_prob = area_above+area_below
print("P(z ≠ 2.07) is ", total_prob)
if total_prob < alpha:
    print("Reject H0")
else:
    print("Fail to reject H0")

P(z ≠ 2.07) is  0.03838242412656269
Reject H0


In [46]:
z_score

-2.070747228595759

In [47]:
z_score1

2.070747228595759

In [53]:
# Alternatively
#divide alpha as 2 tailed test
two_tail_alpha = alpha/2

# compute critcal value
# % area below .025 is 0.975
area = 1-two_tail_alpha
significance_positive_zscore = st.norm.ppf(area)
print("Z_value of {} is : {}".format(area,significance_positive_zscore))
significance_negative_zscore = st.norm.ppf(two_tail_alpha)
print("Z_value of {} is : {}".format(two_tail_alpha,significance_negative_zscore))
print("critical value of significance at 5% is ±{}".format(significance_positive_zscore))

# comparte critical score for accepting and rejecting null hypothesis
if significance_positive_zscore < z_score1 :
    print("Reject H0")
else:
    print("Fail to reject H0")

Z_value of 0.975 is : 1.959963984540054
Z_value of 0.025 is : -1.9599639845400545
critical value of significance at 5% is ±1.959963984540054
Reject H0


### Problem Statement 5:
In a report prepared by the Economic Research Department of a major bank the Department manager maintains that the average annual family income on Metropolis is $48,432. 

What do you conclude about the validity of the report if a random sample of 400 families shows and average income of $48,574 with a standard deviation of 2000?

In [59]:
# lets perform a Z-test
# 𝐻0: 𝜇 = 48432, 𝐻1: 𝜇  ≠  48432
# alpha =10%
actual_mean = 48432
assumed_mean = 48574
n = 400
alpha = 0.10
std_dev = 2000
std_error = std_dev/np.sqrt(n)
z_score = (assumed_mean-actual_mean)/ std_error
print("Z_score : ",z_score)
area_above = 1- st.norm.cdf(z_score)
print("%area above {} is {}".format(z_score,area_above))
z_score1 = st.norm.ppf(area_above)
area_below = st.norm.cdf(z_score1)
print("%area below {} is {}".format(z_score1,area_below))
print("critical value(Z*) is {}, {} ".format(z_score1,z_score))

Z_score :  1.42
%area above 1.42 is 0.07780384052654643
%area below -1.42 is 0.07780384052654638
critical value(Z*) is -1.42, 1.42 


In [60]:
total_prob = area_above+area_below
print("P(z ≠ 1.42) is ", total_prob)
if total_prob < alpha:
    print("Reject H0")
else:
    print("Fail to reject H0")

P(z ≠ 1.42) is  0.1556076810530928
Fail to reject H0


In [61]:
# Alternatively
#divide alpha as 2 tailed test
two_tail_alpha = alpha/2

# compute critcal value
# % area below .025 is 0.975
area = 1-two_tail_alpha
significance_positive_zscore = st.norm.ppf(area)
print("Z_value of {} is : {}".format(area,significance_positive_zscore))
significance_negative_zscore = st.norm.ppf(two_tail_alpha)
print("Z_value of {} is : {}".format(two_tail_alpha,significance_negative_zscore))
print("critical value of significance at 5% is ±{}".format(significance_positive_zscore))

# comparte critical score for accepting and rejecting null hypothesis
if significance_positive_zscore < z_score :
    print("Reject H0")
else:
    print("Fail to reject H0")

Z_value of 0.95 is : 1.6448536269514722
Z_value of 0.05 is : -1.6448536269514729
critical value of significance at 5% is ±1.6448536269514722
Fail to reject H0


### Problem Statement 6:
Suppose that in past years the average price per square foot for warehouses in the United States has been $32.28

A national real estate investor wants to determine whether that figure has changed now. The investor hires a researcher who randomly samples 19 warehouses that are for sale across the United States and finds that the mean price per square foot is $ 31.67,  with a standard deviation of $1.29

assume that the prices of warehouse footage are normally distributed in population. 
If the researcher uses a 5% level of significance, what statistical conclusion can be reached? What are the hypotheses?

In [63]:
# lets perform a Z-test
# 𝐻0: 𝜇 = 32.28, 𝐻1: 𝜇  ≠  31.67
# alpha = 5%
actual_mean = 32.28
assumed_mean = 31.67
n = 19
alpha = 0.05
std_dev = 1.29
std_error = std_dev/np.sqrt(n)
z_score = (assumed_mean-actual_mean)/ std_error
print("Z_score : ",z_score)
area_above = st.norm.cdf(z_score)
print("%area above {} is {}".format(z_score,area_above))
z_score1 = st.norm.ppf(1-area_above)
area_below = 1-st.norm.cdf(z_score1)
print("%area below {} is {}".format(z_score1,area_below))
print("critical value(Z*) is {}, {} ".format(z_score1,z_score))

Z_score :  -2.06118477175179
%area above -2.06118477175179 is 0.019642707888713782
%area below 2.0611847717517895 is 0.019642707888713806
critical value(Z*) is 2.0611847717517895, -2.06118477175179 


In [64]:
total_prob = area_above+area_below
print("P(z ≠ 1.42) is ", total_prob)
if total_prob < alpha:
    print("Reject H0")
else:
    print("Fail to reject H0")

P(z ≠ 1.42) is  0.039285415777427585
Reject H0


In [68]:
# Alternatively
#divide alpha as 2 tailed test
two_tail_alpha = alpha/2

# compute critcal value
# % area below .025 is 0.975
area = 1-two_tail_alpha
significance_positive_zscore = st.norm.ppf(area)
print("Z_value of {} is : {}".format(area,significance_positive_zscore))
significance_negative_zscore = st.norm.ppf(two_tail_alpha)
print("Z_value of {} is : {}".format(two_tail_alpha,significance_negative_zscore))
print("critical value of significance at 5% is ±{}".format(significance_positive_zscore))

# comparte critical score for accepting and rejecting null hypothesis
if significance_positive_zscore < z_score1 :
    print("Reject H0 as ±{} < ±{}".format(significance_positive_zscore,z_score1))
else:
    print("Fail to reject H0 ±{} > ±{}".format(significance_positive_zscore,z_score1))

Z_value of 0.975 is : 1.959963984540054
Z_value of 0.025 is : -1.9599639845400545
critical value of significance at 5% is ±1.959963984540054
Reject H0 as ±1.959963984540054 < ±2.0611847717517895


### Problem Statement 7:
Fill in the blank spaces in the table and draw your conclusions from it.
<img src="img\p7.png">

In [86]:
def compute_prob_interval(mean, x1,x2,sample_size):
    n=sample_size
    std_dev = 4 #since no information is given assueming std_Dev = 4 from better approximations
    se = std_dev/np.sqrt(n)
    z_score1 = (x1-mean)/(se)
    z_score2 = (x2-mean)/(se)
    prob1 = st.norm.cdf(z_score1)
    prob2 = st.norm.cdf(z_score2)
    print("for 𝜇 = {} ".format(mean))
    print("z_score interval({} < x̅ < {}) for ({} < x̅ < {}) ".format(z_score1,z_score2,x1,x2))
    print("probability interval P({} < x̅ < {})".format(prob1,prob2))
    print("Beta region : {} - {} = {}".format(prob2,prob1, prob2-prob1))

In [88]:
compute_prob_interval(52,48.5,51,10)
print()
compute_prob_interval(50.5,48.5,51,10)

for 𝜇 = 52 
z_score interval(-2.766992952647332 < x̅ -0.7905694150420948) for (48.5 < x̅ < 51) 
probability interval P(0.00282879890753218 < x̅ < 0.21459765022017463)
Beta region : 0.21459765022017463 - 0.00282879890753218 = 0.21176885131264245

for 𝜇 = 50.5 
z_score interval(-1.5811388300841895 < x̅ 0.3952847075210474) for (48.5 < x̅ < 51) 
probability interval P(0.056923149003329024 < x̅ < 0.6536836079790199)
Beta region : 0.6536836079790199 - 0.056923149003329024 = 0.5967604589756909


In [89]:
compute_prob_interval(52,48,52,10)
print()
compute_prob_interval(50.5,48,52,10)

for 𝜇 = 52 
z_score interval(-3.162277660168379 < x̅ 0.0) for (48 < x̅ < 52) 
probability interval P(0.000782701129001274 < x̅ < 0.5)
Beta region : 0.5 - 0.000782701129001274 = 0.49921729887099875

for 𝜇 = 50.5 
z_score interval(-1.976423537605237 < x̅ 1.1858541225631423) for (48 < x̅ < 52) 
probability interval P(0.024053413944259697 < x̅ < 0.8821600432854813)
Beta region : 0.8821600432854813 - 0.024053413944259697 = 0.8581066293412215


In [90]:
compute_prob_interval(52,48.81,51.9,16)
print()
compute_prob_interval(50.5,48.81,51.9,16)

for 𝜇 = 52 
z_score interval(-3.1899999999999977 < x̅ -0.10000000000000142) for (48.81 < x̅ < 51.9) 
probability interval P(0.0007113639686453697 < x̅ < 0.46017216272297046)
Beta region : 0.46017216272297046 - 0.0007113639686453697 = 0.4594607987543251

for 𝜇 = 50.5 
z_score interval(-1.6899999999999977 < x̅ 1.3999999999999986) for (48.81 < x̅ < 51.9) 
probability interval P(0.04551397732155001 < x̅ < 0.9192433407662288)
Beta region : 0.9192433407662288 - 0.04551397732155001 = 0.8737293634446788


In [91]:
compute_prob_interval(52,48.42,51.58,16)
print()
compute_prob_interval(50.5,48.42,51.58,16)

for 𝜇 = 52 
z_score interval(-3.5799999999999983 < x̅ -0.4200000000000017) for (48.42 < x̅ < 51.58) 
probability interval P(0.00017179710374593175 < x̅ < 0.33724272684824885)
Beta region : 0.33724272684824885 - 0.00017179710374593175 = 0.3370709297445029

for 𝜇 = 50.5 
z_score interval(-2.0799999999999983 < x̅ 1.0799999999999983) for (48.42 < x̅ < 51.58) 
probability interval P(0.018762766434937836 < x̅ < 0.8599289099112306)
Beta region : 0.8599289099112306 - 0.018762766434937836 = 0.8411661434762928


### Problem Statement 8:
Find the t-score for a sample size of 16 taken from a population with mean 10 when the sample mean is 12 and the sample standard deviation is 1.5.

In [96]:
from scipy.stats import t
n = 16
mean = 10
sample_mean = 12
std_dev = 1.5

se = std_dev/np.sqrt(n)
t_score = (sample_mean-mean)/(se)

print("t_score : ", t_score,"Degrees of Freedom : ",n-1)

t_score :  5.333333333333333 Degrees of Freedom :  15


In [100]:
pval = stats.t.sf(np.abs(t_score), n-1)*2  # two-sided pvalue = Prob(abs(t)>tt)
print('t-statistic = %6.3f pvalue = %6.4f' % (t_score, pval))

t-statistic =  5.333 pvalue = 0.0001


### Problem Statement 9:
Find the t-score below which we can expect 99% of sample means will fall if samples of size 16 are taken from a normally distributed population.

In [101]:
# for small samples (<50) we use t-statistics
# n = 16, degree of freedom = 16-1 = 15
# for 99% confidence interval, alpha = 1% = 0.01 and alpha/2 = 0.005


n= 16
alpha=(1-0.99)/2 # 0.005 is area on both side and area below is .99+.005=.995
print(f"t_score is ",{stats.t.ppf(1-alpha,df=15)})




t_score is  {2.946712883338615}


### Problem Statement 10:
If a random sample of size 25 drawn from a normal population gives a mean of 60 and a standard deviation of 4, find the range of t-scores where we can expect to find the middle 95% of all sample means. Compute the probability that (−𝑡0.05 <𝑡<𝑡0.10).

<img src="img\p10.png">

In [104]:
n= 25
mean = 60
std_dev = 4
se = std_dev/np.sqrt(25)
alpha=(1-0.95)/2 # 0.025 is area on both side and area below is .95+.025=.975
t_score = stats.t.ppf(1-alpha,df=24)
print("t_score is ", t_score)

t_score is  2.0638985616280205


In [107]:
print("Confidence Interval : ",mean-t_score*se," , ",mean+t_score*se)

Confidence Interval :  58.348881150697586  ,  61.651118849302414


In [110]:
prob1 = stats.t.cdf(0.1,df=24)
prob2 = stats.t.cdf(-0.05,df=24)
print("P(−𝑡0.05 <𝑡<𝑡0.10) : ",prob1-prob2)

P(−𝑡0.05 <𝑡<𝑡0.10) :  0.05914441613731247


### Problem Statement 11:
Two-tailed test for difference between two population means. Is there evidence to conclude that the number of people travelling from Bangalore to Chennai is different from the number of people travelling from Bangalore to Hosur in
a week, given the following:

Population 1: Bangalore to Chennai n1 = 1200

x1 = 452

s1 = 212

Population 2: Bangalore to Hosur n2 = 800

x2 = 523

s2 = 185

In [129]:
# H0: there is no difference between mean
# H1: there is difference between mean
n1 = 1200
x1 = 452
s1 = 212

n2 = 800
x2 = 523
s2 = 185

SE = np.sqrt(((s1**2)/n1) + ((s2**2)/n2)) 
z_score = (x1-x2) / SE
print("Z_Score is", z_score)

Z_Score is -7.926428526759299


In [134]:
alpha = 0.05 #or 5%
two_tail_alpha = alpha/2
# compute critcal value
# % area below .025 is 0.975
area = 1-two_tail_alpha
significance_positive_zscore = st.norm.ppf(area)
print("Z_value of {} is : {}".format(area,significance_positive_zscore))
significance_negative_zscore = st.norm.ppf(two_tail_alpha)
print("Z_value of {} is : {}".format(two_tail_alpha,significance_negative_zscore))
print("critical value of significance at 5% is ±{}".format(significance_positive_zscore))

# comparte critical score for accepting and rejecting null hypothesis
if significance_negative_zscore < z_score :
    print("Reject H0 as ±{} < {}".format(significance_positive_zscore,z_score))
else:
    print("Fail to reject H0 ±{} < {}".format(significance_positive_zscore,z_score))

Z_value of 0.975 is : 1.959963984540054
Z_value of 0.025 is : -1.9599639845400545
critical value of significance at 5% is ±1.959963984540054
Reject H0 as ±1.959963984540054 < 5.025702668336442


### Problem Statement 12:
Is there evidence to conclude that the number of people preferring Duracell battery is different from the number of people preferring Energizer battery, given the following:

Population 1: Duracell

n1 = 100

x1 = 308

s1 = 84

Population 2: Energizer

n2 = 100

x2 = 254

s2 = 67

In [131]:
# H0: there is no difference between duracell and Energizer
# H1: there is difference between duracell and Energizer
n1 = 100
x1 = 308
s1 = 84

n2 = 100
x2 = 254
s2 = 67

SE = np.sqrt(((s1**2)/n1) + ((s2**2)/n2)) 
z_score = (x1-x2) / SE
print("Z_Score is", z_score)

Z_Score is 5.025702668336442


In [133]:
# compute critcal value
# % area below .025 is 0.975
area = 1-two_tail_alpha
significance_positive_zscore = st.norm.ppf(area)
print("Z_value of {} is : {}".format(area,significance_positive_zscore))
significance_negative_zscore = st.norm.ppf(two_tail_alpha)
print("Z_value of {} is : {}".format(two_tail_alpha,significance_negative_zscore))
print("critical value of significance at 5% is ±{}".format(significance_positive_zscore))

# comparte critical score for accepting and rejecting null hypothesis
if  significance_positive_zscore < z_score   :
    print("Reject H0 as ±{} < {}".format(significance_positive_zscore,z_score))
else:
    print("Fail to reject H0 ±{} > {}".format(significance_positive_zscore,z_score))

Z_value of 0.975 is : 1.959963984540054
Z_value of 0.025 is : -1.9599639845400545
critical value of significance at 5% is ±1.959963984540054
Reject H0 as ±1.959963984540054 < 5.025702668336442


### Problem Statement 13:
Pooled estimate of the population variance. Does the data provide sufficient evidence to conclude that average percentage
increase in the price of sugar differs when it is sold at two different prices?

Population 1: Price of sugar = Rs. 27.50 

n1 = 14

x1 = 0.317%

s1 = 0.12%

Population 2: Price of sugar = Rs. 20.00 

n2 = 9

x2 = 0.21%

s2 = 0.11%

https://www.slideshare.net/ShakeelNouman1/the-comparison-of-two-populations

<img src="img\p13.png">


<img src="img\p13-1.png">

In [136]:
# H0: there is no difference between duracell and Energizer
# H1: there is difference between duracell and Energizer
n1 = 14
x1 = .317
s1 = .12

n2 = 9
x2 = .21
s2 = .11

df = n1+n2-2
sample_varience = (((n1-1)*(s1**2)) + ((n2-1)*(s2**2)))/df
t_score = (x1-x2)/np.sqrt(sample_varience * ((1/n1)+(1/n2)))
t_score

2.15355322387416

In [139]:
# Lets test the t_score with significance level 0.05
alpha = 0.05 #or 5%
two_tail_alpha = alpha/2
# compute critcal value
# % area below .025 is 0.975
area = 1-two_tail_alpha
t_score = stats.t.ppf(1-alpha,df=24)
print("critical t_score is ", t_score)

critical t_score is  1.7108820799094275


1.71<2.15 which say that obtained t_score is above the level of significance hence we 'reject the Null Hypothesis'

### Problem Statement 14:
The manufacturers of compact disk players want to test whether a small price reduction is enough to increase sales of their product. Is there evidence that the small price reduction is enough to increase sales of compact disk players?

Population 1: Before reduction

n1 = 15

x1 = Rs. 6598 

s1 = Rs. 844

Population 2: After reduction 

n2 = 12

x2 = RS. 6870

s2 = Rs. 669

In [140]:
# H0: there is no difference between duracell and Energizer
# H1: there is difference between duracell and Energizer
n1 = 15
x1 = 6598
s1 = 844

n2 = 12
x2 = 6870
s2 = 669

df = n1+n2-2
sample_varience = (((n1-1)*(s1**2)) + ((n2-1)*(s2**2)))/df
t_score = (x1-x2)/np.sqrt(sample_varience * ((1/n1)+(1/n2)))
t_score

-0.9098300343990461

In [142]:
# Lets test the t_score with significance level 0.05
alpha = 0.05 #or 5%
area = 1-two_tail_alpha
t_score = stats.t.ppf(alpha,df=26)
print("critical t_score is ", t_score)

critical t_score is  -1.7056179197592731


-1.70 < -0.91 hence we accept the 'Null hypothesis'

### Problem Statement 15:
Comparisons of two population proportions when the hypothesized difference is zero Carry out a two-tailed test of the equality of banks’ share of the car loan market in 1980 and 1995.

Population 1: 1980

n1 = 1000

x1 = 53

𝑝1 = 0.53

Population 2: 1985

n2 = 100

x2 = 43

𝑝 2= 0.43

In [156]:
# H0: there is no difference between duracell and Energizer
# H1: there is difference between duracell and Energizer
n1 = 100
x1 = 53
𝑝1 = 0.53

n2 = 100
x2 = 43
𝑝2= 0.43

p_hat = (x1+x2)/(n1+n2)

z_score = (p1-p2)/np.sqrt((p_hat*(1-p_hat)) * ((1/n1)+(1/n2)))
z_score

1.4153462926807452

In [159]:
alpha = 0.10 #or 10% significance
two_tail_alpha = alpha/2
# compute critcal value
# % area below .05 is 0.95
area = 1-two_tail_alpha
significance_positive_zscore = st.norm.ppf(area)
print("Z_value of {} is : {}".format(area,significance_positive_zscore))
significance_negative_zscore = st.norm.ppf(two_tail_alpha)
print("Z_value of {} is : {}".format(two_tail_alpha,significance_negative_zscore))
print("critical value of significance at 5% is ±{}".format(significance_positive_zscore))

# comparte critical score for accepting and rejecting null hypothesis
if significance_positive_zscore < z_score :
    print("Reject H0 as ±{} < {}".format(significance_positive_zscore,z_score))
else:
    print("Fail to reject H0 ±{} > {}".format(significance_positive_zscore,z_score))

Z_value of 0.95 is : 1.6448536269514722
Z_value of 0.05 is : -1.6448536269514729
critical value of significance at 5% is ±1.6448536269514722
Fail to reject H0 ±1.6448536269514722 < 1.4153462926807452


### Problem Statement 16:
Carry out a one-tailed test to determine whether the population proportion of traveler’s check buyers who buy at least $2500 in checks when sweepstakes prizes are offered as at least 10% higher than the proportion of such buyers when no sweepstakes are on.

Population 1: With sweepstakes

n1 = 300

x1 = 120

𝑝1 = 0.40

Population 2: No sweepstakes 

n2 = 700

x2 = 140

𝑝2= 0.20

In [2]:
#h0 : p1-p2<=0.10
#h1 : p1-p2>0.10

n1 = 300
x1 = 120
p1 = 0.40

n2 = 700
x2 = 140
p2= 0.20

p_hat = (x1+x2)/(n1+n2)

z_score = (p1-p2-0.1)/np.sqrt((p_hat*(1-p_hat)) * ((1/n1)+(1/n2)))
z_score

3.303749523611152

In [6]:
st.norm.cdf(z_score)

0.9995229948610035

In [7]:
area_below = st.norm.cdf(z_score)
area_above = 1-area_below
print("area below z-socre {} is {} and area above is {}".format(z_score,area_below,area_above))

area below z-socre 3.303749523611152 is 0.9995229948610035 and area above is 0.0004770051389965113


In [9]:
alpha = 0.10
if alpha < area_above:
    print("fail to reject H0")
else:
    print("Reject H0")
    

Reject H0


### Problem Statement 17:
A die is thrown 132 times with the following results: Number turned up: 1, 2, 3, 4, 5, 6

Frequency: 16, 20, 25, 14, 29, 28

Is the die unbiased? Consider the degrees of freedom as 𝑝 − 1 .

<img src="img\p17.png">

In [17]:
expected = [22,22,22,22,22,22]
observed = [16,20,25,14,29,28]
X = []
for i in range(6):
    r = ((observed[i] - expected[i])**2)/expected[i]
    X.append(r)
chi_score = np.sum(X)
df = 6-1

In [25]:
from scipy.stats.distributions import chi2
pval = chi2.sf(chi_score,df)
print("P(X={}) = {}".format(chi_score,pval))

P(X=9.0) = 0.1090641579497725


In [26]:
alpha = .10
if pval > alpha:
    print("fail to reject H0")
else:
    print("reject H0")

fail to reject H0


### Problem Statement 18:
In a certain town, there are about one million eligible voters. A simple random sample of 10,000 eligible voters was chosen to study the relationship between gender and participation in the last election. The results are summarized in the following 2X2 (read two by two) contingency table:
<img src="img\p18.png">

<img src="img\p18-1.png">

In [27]:
def compute_expected(val1,val2,total):
    expected = (val1*val2)/total
    return expected

In [45]:
o_men_v = 2792
o_men_nv = 1486
o_women_v = 3591
o_women_nv = 2131
all_voted = o_men_v + o_women_v
not_voted = o_men_nv + o_women_nv
total_men = o_men_v+o_men_nv
total_women = o_women_v+o_women_nv
total = o_men_v+o_men_nv+o_women_v+o_women_nv

e_men_v = compute_expected(all_voted,total_men,total)
e_men_nv = compute_expected(not_voted,total_men,total)
e_women_v = compute_expected(all_voted,total_women,total)
e_women_nv = compute_expected(not_voted,total_women,total)

observed_list = [o_men_v,o_men_nv,o_women_v,o_women_nv]
expected_list = [round(e_men_v),round(e_men_nv),round(e_women_v),round(e_women_nv)]

In [46]:
observed_list

[2792, 1486, 3591, 2131]

In [47]:
expected_list

[2731, 1547, 3652, 2070]

In [48]:
X = []
for i in range(len(observed_list)):
    r = ((observed_list[i] - expected_list[i])**2)/expected_list[i]
    X.append(r)
chi_score = np.sum(X)
df = (2-1)*(2-1)

In [49]:
chi_score

6.58428345675753

In [50]:
pval = chi2.sf(chi_score,df)
print("P(X={}) = {}".format(chi_score,pval))

P(X=6.58428345675753) = 0.010288302079717253


In [51]:
alpha = .10
if pval > alpha:
    print("fail to reject H0")
else:
    print("reject H0")

reject H0


### Problem Statement 19:
A sample of 100 voters are asked which of four candidates they would vote for in an election. The number supporting each candidate is given below:
<img src="img\p19.png">
Do the data suggest that all candidates are equally popular? [Chi-Square = 14.96,with 3 df, 𝑝 0.05 .



<img src="img\p19-1.png">

In [56]:
pval = chi2.sf(14.93,3)
print("P(X={}) = {}".format(14.93,pval))

P(X=14.93) = 0.0018774564613254039


In [57]:
alpha = .05
if pval > alpha:
    print("fail to reject H0")
else:
    print("reject H0")

reject H0


### Problem Statement 20:
Children of three ages are asked to indicate their preference for three photographs of adults. Do the data suggest that there is a significant relationship between age and photograph preference? What is wrong with this study? [Chi-Square = 29.6, with 4 df: 𝑝 < 0.05].
<img src="img\p20.png">

In [58]:
pval = chi2.sf(29.6,4)
print("P(X={}) = {}".format(29.6,pval))

P(X=29.6) = 5.903353020218714e-06


In [59]:
alpha = .05
if pval > alpha:
    print("fail to reject H0")
else:
    print("reject H0")

reject H0


### Problem Statement 21:
A study of conformity using the Asch paradigm involved two conditions: one where one confederate supported the true judgement and another where no confederate gave the correct response.
<img src="img\p21.png">
Is there a significant difference between the "support" and "no support" conditions in the frequency with which individuals are likely to conform? [Chi-Square = 19.87, with 1 df: 𝑝 < 0.05].

In [60]:
pval = chi2.sf(19.87,1)
print("P(X={}) = {}".format(19.87,pval))

P(X=19.87) = 8.289095026714455e-06


In [61]:
alpha = .05
if pval > alpha:
    print("fail to reject H0")
else:
    print("reject H0")

reject H0


## Problem Statement 22:
We want to test whether short people differ with respect to their leadership qualities (Genghis Khan, Adolf Hitler and Napoleon were all stature-deprived, and how many midget MP's are there?) The following table shows the frequencies with which 43 short people and 52 tall people were categorized as "leaders", "followers" or as "unclassifiable". Is there a relationship between height and leadership qualities?
[Chi-Square = 10.71, with 2 df: 𝑝 < 0.01].
<img src="img\p22.png">

In [62]:
pval = chi2.sf(10.71,2)
print("P(X={}) = {}".format(10.71,pval))

P(X=10.71) = 0.0047244694975056255


In [63]:
alpha = .05
if pval > alpha:
    print("fail to reject H0")
else:
    print("reject H0")

reject H0


### Problem Statement 23:
Each respondent in the Current Population Survey of March 1993 was classified as employed, unemployed, or outside the labor force. The results for men in California age 35-44 can be cross-tabulated by marital status, as follows:
<img src="img\p23.png">
Men of different marital status seem to have different distributions of labor force status. Or is this just chance variation? (you may assume the table results from a simple random sample.)

In [ ]:
def compute_expected(val1,val2,total):
    expected = (val1*val2)/total
    return expected

In [65]:
o_married_e = 679
o_married_u = 63
o_married_nlf = 42

o_widowed_e = 103
o_widowed_u = 10
o_widowed_nlf = 18

o_nvr_married_e = 114
o_nvr_married_u = 20
o_nvr_married_nlf = 25

all_employed = o_married_e+o_widowed_e+o_nvr_married_e
all_unemployed = o_married_u+o_widowed_u+o_nvr_married_u
all_nlf = o_married_nlf+o_widowed_nlf+o_nvr_married_nlf

all_married = o_married_e+o_married_u+o_married_nlf
all_widowed = o_widowed_e+o_widowed_u+o_widowed_nlf
all_nvr_married = o_nvr_married_e+o_nvr_married_u+o_nvr_married_nlf

total = o_married_e+o_married_u+o_married_nlf+o_widowed_e+o_widowed_u+o_widowed_nlf+o_nvr_married_e+o_nvr_married_u+o_nvr_married_nlf

In [66]:
e_married_e = compute_expected(all_employed,all_married,total)
e_married_u = compute_expected(all_unemployed,all_married,total)
e_married_nlf = compute_expected(all_nlf,all_married,total)

e_widowed_u = compute_expected(all_employed,all_widowed,total)
e_nvr_married_e = compute_expected(all_employed,all_nvr_married,total)


e_men_nv = compute_expected(all_unemployed,all_widowed,total)
e_women_v = compute_expected(all_unemployed,all_nlf,total)


e_men_nv = compute_expected(all_nlf,all_widowed,total)
e_women_v = compute_expected(all_nlf,all_nlf,total)



observed_list = [o_men_v,o_men_nv,o_women_v,o_women_nv]
expected_list = [round(e_men_v),round(e_men_nv),round(e_women_v),round(e_women_nv)]

1074